meson.build

```diff
@@ -42,141 +42,152 @@ pkg.generate(
+		'unit/test-aaf.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies: cmocka,
+		build_by_default: false,
+	)
+
+	test_crf = executable(
+		'test-crf',
+		'unit/test-crf.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies: cmocka,
+		build_by_default: false,
+	)
+
+	test_stream = executable(
+		'test-stream',
+		'unit/test-stream.c',
+		'src/avtp_stream.c',
+		include_directories: include_directories('include', 'src'),
+		link_with: avtp_lib,
+		dependencies: cmocka,
+		build_by_default: false,
+	)
+
+	test_cvf = executable(
+		'test-cvf',
+		'unit/test-cvf.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies: cmocka,
+		build_by_default: false,
+	)
+
+	test_ieciidc = executable(
+		'test-ieciidc',
+		'unit/test-ieciidc.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies: cmocka,
+		build_by_default: false,
+	)
+
+	test('AVTP API', test_avtp)
+	test('Stream API', test_stream)
+	test('AAF API', test_aaf)
+	test('CRF API', test_crf)
+	test('CVF API', test_cvf)
+	test('IEC61883/IIDC API', test_ieciidc)
+endif
+
+if get_option('tests') != 'disabled'
+
+	cc = meson.get_compiler('c')
+	mdep = cc.find_library('m', required : false)
+
+	executable(
+		'aaf-talker',
+		'examples/aaf-talker.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+
+	executable(
+		'aaf-listener',
+		'examples/aaf-listener.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+
+	executable(
+		'crf-talker',
+		'examples/crf-talker.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies : mdep,
+		build_by_default: false,
+	)
+
+	executable(
+		'crf-listener',
+		'examples/crf-listener.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		dependencies : mdep,
+		build_by_default: false,
+	)
+
+	executable(
+		'cvf-talker',
+		'examples/cvf-talker.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+
+	executable(
+		'cvf-listener',
+		'examples/cvf-listener.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+
+	executable(
+		'ieciidc-talker',
+		'examples/ieciidc-talker.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+
+	executable(
+		'ieciidc-listener',
+		'examples/ieciidc-listener.c',
+		'examples/common.c',
+		include_directories: include_directories('include'),
+		link_with: avtp_lib,
+		build_by_default: false,
+	)
+endif
```

Comment:
This comment is related to the whole patch: Since cmocka is only required when building unit tests, I'm wondering why not simply defining a variable `cmocka_dep = dependency('cmocka', required: false)' and use it in test_* executables.

Reply:
It is cleaner to add options for optional dependencies.